lib/awful/tag.lua

```diff
@@ -1355,6 +1355,32 @@ capi.screen.add_signal("tag::history::update")
 
 capi.screen.connect_signal("tag::history::update", tag.history.update)
 
+capi.screen.connect_signal("removed", function(s)
+    local old_tags = s.tags
+    -- First try to get delete tags gently
+    for _, t in pairs(s.tags) do
+        t:delete()
+    end
+    -- Then be a little more fierce and force clients to go somewhere random
+    local fallback = nil
+    for other_screen in capi.screen do
+        if #other_screen.tags > 0 then
+            fallback = other_screen.tags[1]
```

Comment:
This should be done in a request::screen handler. Tyrannical may want to move "volatile" tags to existing screens instead of just deleting them. This is also true for the `:delete()` above. The default handler should delete the tags. This make total sense for the default config where tags are less or more used as virtual workspaces.